tests/unit_tests/query_handler/test_scope_query_handler_context.py

```diff
@@ -284,18 +286,52 @@ def test_release_parent_before_child_with_temporary_object_expect_exception(
     parent = scope_query_handler_context
     child = scope_query_handler_context.get_child_query_handler_context()
     proxy = child.get_temporary_table_name()
-    with pytest.raises(RuntimeError, match=f"Child contexts are not released."):
+    with pytest.raises(ChildContextNotReleasedError):
         parent.release()
 
 
 def test_release_parent_before_child_without_temporary_object_expect_exception(
         scope_query_handler_context: ScopeQueryHandlerContext):
     parent = scope_query_handler_context
     child = scope_query_handler_context.get_child_query_handler_context()
-    with pytest.raises(RuntimeError, match=f"Child contexts are not released."):
+    with pytest.raises(ChildContextNotReleasedError):
         parent.release()
 
 
+def test_release_parent_before_grand_child_with_temporary_object_expect_exception(
+        scope_query_handler_context: ScopeQueryHandlerContext):
+    parent = scope_query_handler_context
+    child = scope_query_handler_context.get_child_query_handler_context()
+    grand_child = child.get_child_query_handler_context()
+    proxy = grand_child.get_temporary_table_name()
```

Comment:
Do you need the assignment here or can you simply call the method without the assignment? Or would the object then go out of scope and be remove automatically?

Reply:
Without assignment, the style checker complains

Reply:
:-)

Okay.